File: 20160305_P1.java
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;

public class P1 {

    //for input
    private static Scanner scanner;

    //number of students
    private int n;

    //container for passed students
    private List<Integer> pass;

    //container for failed students
    private List<Integer> fail;

    //print list with space as delimiter
    private void printList(List<Integer> list) {

        for (int i = 0; i < list.size(); i++) {
            if (i != 0) {
                System.out.print(" ");
            }
            System.out.print(list.get(i));
        }
    }

    //constructor
    P1 () {
        //initial a Scanner object using keyboard as input
        scanner = new Scanner(System.in);

        //get an integer from keyboard
        n = scanner.nextInt();

        //initial two integer list to store grades
        pass = new ArrayList<Integer>(n);
        fail = new ArrayList<Integer>(n);

        //get the list ready
        for (int i = 0; i < n; i++) {
            int temp = scanner.nextInt();
            if (temp < 60) {
                fail.add(temp);
            } else {
                pass.add(temp);
            }
        }

        //sort two list as natural ordering
        pass.sort(null);
        fail.sort(null);

        //print all grades
        printList(fail);
        System.out.print(" ");
        printList(pass);
        System.out.println();

        //no students are failed if fail is empty
        if (fail.isEmpty()) {
            System.out.println("best case");
        } else {
            //print out the largest grade of fail
            System.out.println(fail.get(fail.size()-1));
        }

        //no students are passed if pass is empty
        if (pass.isEmpty()) {
            System.out.println("worst case");
        } else {
            //print out the smallest grade of pass
            System.out.println(pass.get(0));
        }
    }

    //main function
    public static void main(String... arg) {
        //new a object to trigger constructor
        new P1();
    }
}

File: 20160305_P1.cpp
#include <iostream>
#include <vector>
#include <algorithm> //std::sort

//print vector v with space as delimiter
void printVector (std::vector<int> v) {

    for (int i = 0; i < v.size(); i++) {
        if (i) {
            std::cout << " ";
        }
        std::cout << v[i];
    }
}

int main() {

    //number of students
    int n;
    std::cin >> n;

    //container for failed students
    std::vector<int> failed;

    //container for passed students
    std::vector<int> passed;

    //fetch inputs
    for (int i = 0; i < n; i++) {
        int tmp;
        std::cin >> tmp;
        if (tmp < 60) {
            failed.push_back(tmp);
        } else {
            passed.push_back(tmp);
        }
    }

    //sort two vector
    std::sort(failed.begin(), failed.end());
    std::sort(passed.begin(), passed.end());

    //print all grades
    printVector(failed);
    std::cout << " ";
    printVector(passed);
    std::cout << std::endl;

    //no students are failed if fail is empty
    if (failed.empty()) {
        std::cout << "best case" << std::endl;
    } else {
        //print out the largest grade of fail
        std::cout << failed[failed.size()-1] << std::endl;
    }

    //no students are passed if pass is empty
    if (passed.empty()) {
        std::cout << "worst case" << std::endl;
    } else {
        //print out the smallest grade of pass
        std::cout << passed[0] << std::endl;
    }
}
